Support custom SSL certificates for the dashboard

benniekiss opened the pull request:
This PR refactors the init_cert.sh script in the docker container to support using custom SSL certificates.

I've added two env-vars, `NGINX_SSL_CERT` and `NGINX_SSL_KEY`, which when set, will configure the nginx default.conf with the provided port and certificate paths. When `LETSENCRYPT_DOMAIN` is set, `NGINX_SSL_CERT` and `NGINX_SSL_KEY` are ignored.

I've also made it so that the dashboard port can be changed without providing certificates. This way, setting `NGINX_SSL_PORT` will change the nginx listen port. It may be worth considering changing the env-var name. 

This partially closes https://github.com/netbirdio/netbird/issues/2242

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/netbirdio/dashboard?pullRequest=399) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/netbirdio/dashboard?pullRequest=399) before we can accept your contribution.<br/><hr/>**bgmt** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/netbirdio/dashboard?pullRequest=399) it.</sub>

benniekiss commented:
Force-pushed to fix authorship

iammuho commented:
When that PR will be merged ?

benniekiss commented:
This PR needs updated to match the correct format for the `http2` directive. I'll try to get to that when I can, but I've since moved to a custom dashboard container, so this is not really on my priority list.

benniekiss commented:
I've updated the `http2` directive and updated the alpine version (it was very very old!). I haven't run any tests since I do not use the regular dashboard container in my deployment anymore, so if a community member wants to test and ensure this works, please do! I believe it should.

@bcmmbaga @mlsmaycon this is ready for review if you are interested in taking a look.
